### ISSUE TITLE
Test and debug the initial packer builds and images

### ISSUE THREAD

**Issue by dongsupark** [flatcar-linux/flatcar-packer-qemu#2]
I did an initial work on https://github.com/flatcar-linux/flatcar-packer-qemu/pull/1 to be able to support libvirt vagrant provider for Flatcar Linux. This PR is basically a rewrite based on existing open source projects, such as https://github.com/dyson/packer-qemu-coreos-container-linux or https://github.com/bfraser/packer-coreos-qemu. The result image for the alpha channel is also available on https://app.vagrantup.com/dongsupark/boxes/flatcar-alpha.

It seems to work for now. Though there are still some things to be done.

- [x] Find (and/or) fix bugs in the packer build. For example:
  - We need to figure out a better way to define the ssh public key defined in `flatcar-linux.json` or in `ignition.json`. This part looks like a hack.
  - **(Decided to do this afterwards)** We need to host the images on `vagrantup.com` with our own username `flatcar`, instead of a personal account like `dongsupark`.
  - I needed to set `keep_input_artifact` to `true` in the `post-processors` section, to avoid issues of result images disappearing in the end. This part looks weird, but it doesn't work without the option. I'm not sure if it's an upstream packer bug, or an expected behavior. Or there might be a better way to deal with the issue.
  - **(DONE, merged to upstream)** I already fixed an issue of boot failures by increasing RAM to 2G, and created [an upstream PR](https://github.com/dyson/packer-qemu-coreos-container-linux/pull/1). We need to follow and watch it in the future.
  - Let's go through other configurations defined in `flatcar-linux.json` or `ignition.json`, and figure out if they are really needed. As most of them are copied from other projects, I'm not sure if every defined config is actually necessary.
- [x] Test result vagrant images, by running `vagrant up --provider=libvirt` on a vagrant cluster.
- [x] Support other channels, `beta` and `stable`. This can be done only after beta and stable images would be available on our Flatcar web server.

/cc @kosyfrances

NOTE: this issue is the same as https://github.com/kinvolk/flatcar-packer-qemu/issues/2. The only difference is the new organization name `flatcar-linux`.

**Comment by kosyfrances** [comment]
```I needed to set keep_input_artifact to true in the post-processors section, to avoid issues of result images disappearing in the end. This part looks weird, but it doesn't work without the option. I'm not sure if it's an upstream packer bug, or an expected behavior. Or there might be a better way to deal with the issue.```

This seems to be the expected behaviour according to what I understand in the explanation here https://www.packer.io/docs/templates/post-processors.html#input-artifacts. The end goal will be to push the resulting image to  https://app.vagrantup.com right? If yes, we can decide whether to keep the artifact locally or just push directly to vagrantup as soon as the process is done without keeping any build locally. What do you think @dongsupark ?

**Comment by dongsupark** [comment]
@kosyfrances Yes, that's true.
What I meant is like this:
If `keep_input_artifact` is set to false (which is default), and if `post-processors` has no entry `atlas` like the one defined in https://github.com/bfraser/packer-coreos-qemu/blob/master/coreos-qemu.json#L72..L81, then I would expect the result files to be stored locally, without being deleted. However actually it was not the case. The output `builds` directory was silently deleted, without being uploaded, and I could not find any file. That looked quite strange to me.

Anyway I don't care much about that behavior. I'm fine if the output files are correctly generated, no matter whether it's locally stored or uploaded. Feel free to skip that issue. :-)

**Comment by kosyfrances** [comment]
@dongsupark Do I need to upload the boxes for beta and stable or just modify the code to support both channels?

**Comment by dongsupark** [comment]
@kosyfrances Just modify the code. Then I can build the images and upload them on my own.

**Comment by dongsupark** [comment]
Every major part is done. For minor things left, let's create another issue.
Thanks!
